### PR TITLE
feature: conditions calling base

### PR DESCRIPTION
**Pull request by klakpin**
## Description

Please carefully read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md) before making any pull requests.
And, **Do not make a pull request to merge into stable unless it is a hotfix. Use the development branch instead.**
## Type of changes

- [ ] Bug fix
- [ ] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My pull request is unique and no other pull requests have been opened for these changes
- [ ] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [ ] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: src/main/java/emu/grasscutter/game/activity/condition/PlayerActivityDataMappingBuilder.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+     *     <li>Activity data id one of [1001, 1002]. Then cond id will be activity data id plus 2 additional digits.
+     *     For example: activity data id = 1001, cond id = 1001x (x - any digit></li>
+     * </ol>
+     *
+     * @param key cond id for which activity data id should be defined
+     * @return activity data for given cond id. Returns -1 if activity was not found.
+     */
+    private Integer detectActivityDataIdByCondId(Integer key) {
+        if (key / 10 == 1001 || key / 10 == 1002) {
+            return key / 10;
```
Comment (Hartie95) [inline src/main/java/emu/grasscutter/game/activity/condition/PlayerActivityDataMappingBuilder.java:73]: both are 1001, so you can probably just return 1001 here

File: src/main/java/emu/grasscutter/game/activity/condition/all/DaysGreatEqual.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import java.util.Date;
+
+import static emu.grasscutter.game.activity.condition.ActivityConditions.NEW_ACTIVITY_COND_DAYS_GREAT_EQUAL;
+
+@ActivityCondition(NEW_ACTIVITY_COND_DAYS_GREAT_EQUAL)
+public class DaysGreatEqual extends ActivityConditionBaseHandler {
+    @Override
+    public boolean execute(PlayerActivityData activityData, ActivityConfigItem activityConfig, int... params) {
+        Date activityBeginTime = activityConfig.getBeginTime();
```
Comment (Hartie95) [inline src/main/java/emu/grasscutter/game/activity/condition/all/DaysGreatEqual.java:16]: maybe it could be good to also test that begintime is after now

Comment (Hartie95) [reply src/main/java/emu/grasscutter/game/activity/condition/all/DaysGreatEqual.java:16]: When it's late while reviewing and I don't notice it's negative in that case xD

File: src/main/java/emu/grasscutter/game/activity/condition/PlayerActivityDataMappingBuilder.java (hunk 1 above)
Comment (klakpin) [reply src/main/java/emu/grasscutter/game/activity/condition/PlayerActivityDataMappingBuilder.java]: Fixed

**Pull request merged**